api/apikey/cache.go

```diff
@@ -55,3 +55,14 @@ func (c *apiKeyCache) Set(digest []byte, user portainer.User, apiKey portainer.A
 func (c *apiKeyCache) Delete(digest []byte) {
 	c.cache.Remove(string(digest))
 }
+
+func (c *apiKeyCache) InvalidateUserKeyCache(userId portainer.UserID) bool {
+	present := false
+	for _, k := range c.cache.Keys() {
+		user, _, _ := c.Get([]byte(k.(string)))
+		if user.ID == userId {
+			present = c.cache.Remove(k)
```

Comment:
Wondering if the key can exist only once or multiple times.  If only once then why not `return c.cache.Remove(k)`

Reply:
Users could have multiple keys - we need to loop through all the keys to remove them.